services/workflows-service/src/data-analytics/data-analytics.service.ts

```diff
@@ -13,41 +13,34 @@ import { isEmpty } from 'lodash';
 
 @Injectable()
 export class DataAnalyticsService {
-  private _evaluateNameToFunction: Partial<{
-    [K in keyof EvaluateFunctions]: EvaluateFunctions[K];
-  }> = {};
-
   constructor(
     protected readonly prisma: PrismaService,
     protected readonly logger: AppLoggerService,
-  ) {
-    this._evaluateNameToFunction[
-      this.evaluateTransactionsAgainstDynamicRules.name as keyof EvaluateFunctions
-    ] = this.evaluateTransactionsAgainstDynamicRules.bind(this);
-
-    this._evaluateNameToFunction[
-      this.evaluateCustomersTransactionType.name as keyof EvaluateFunctions
-    ] = this.evaluateCustomersTransactionType.bind(this);
-
-    // this._evaluateNameToFunction[this.evaluateDormantAccount.name] =
-    //   this.evaluateDormantAccount.bind(this);
-  }
-
-  async runInlineRule(projectId: string, inlineRule: InlineRule) {
-    const evaluateFn = this._evaluateNameToFunction[inlineRule.fnName as keyof EvaluateFunctions];
-
-    if (!evaluateFn) {
-      this.logger.error(`No evaluation function found for rule name: ${inlineRule.id}`, {
-        inlineRule,
-      });
-
-      throw new Error(`No evaluation function found for rule name: ${inlineRule.id}`);
+  ) {}
+
+  async runInlineRule(projectId: string, { fnName, options }: InlineRule) {
```

Comment:
You have removed the logger whenever a evaluate fn doesnt exists

Reply:
Fixed in next commit